packages/app/src/connection-form.tsx

```diff
@@ -15,7 +15,7 @@ export function ConnectionForm({
   submitButton,
 }: {
   onSubmit: (values: z.infer<typeof connectionFormSchema>) => void
-  defaultValues: z.infer<typeof connectionFormSchema>
```

Comment:
I think we want to keep the z.infer here so the values are type safe. AppForm has type of `any` according to TS on my computer, so using it to extract ComponentProps defaultValues probably isn't working

Reply:
Ideally, this should be of whatever AppForm type is. Unfortunately, it's `any`. `z.infer<typeof connectionFormSchema>` was a bit misleading, because if `connectionFormSchema` was incompatible with the AppForm, it would still satisfy TypeScript type checking if you passed an instance of it. 

Either way, it's not really type safe, but I don't find it a big issue given it's just a single form at this point. I am happy to make adjustments if you feel strong about that one tho!